src/Options/index.js

```diff
@@ -149,7 +149,7 @@ export interface ParseServerOptions {
   allowCustomObjectId: ?boolean;
   /* Configuration for your authentication providers, as stringified JSON. See http://docs.parseplatform.org/parse-server/guide/#oauth-and-3rd-party-authentication
   :ENV: PARSE_SERVER_AUTH_PROVIDERS */
-  auth: ?(AuthAdapter[]);
+  auth: ?{ [string]: AuthAdapter };
```

Comment:
It's meant to be the format:

```js
auth: {
  totp: {
    enabled: true,
    validateLogin() 
  }
}
```

How would you write it?

Reply:
Got it now (related: https://github.com/parse-community/parse-server/pull/8535#discussion_r1198492032) Is this a breaking change though? I assume the server would check the types provided to the `auth` option and wouldn't accept the old type anymore?

Reply:
So as far as I can tell, the definitions are only used and validated automatically when Parse Server is started by the CLI (maybe we should address this in #8202).

When the auth was changed to an array of auth adapters, the error was not picked up because there isn't any tests to ensure each specific config option is correctly passed through. This was the bug that broke auth adapters for starting via CLI.

Parse Server accesses auth via `auth.facebook` for example, so setting an array of auth adapters wouldn't work, as the array wouldn't have the property "facebook".

So:

- Setting an object of auth adapters wouldn't work before this change (the bug)
- Setting an array of auth adapters wouldn't work before this change (the incorrect change)
- Setting an array of auth adapters wouldn't work after this change (it was never supported)